src/main/java/rx/observables/BlockingSingle.java

```diff
@@ -0,0 +1,89 @@
+package rx.observables;
+
+import rx.Single;
+import rx.SingleSubscriber;
+import rx.Subscription;
+import rx.annotations.Experimental;
+import rx.internal.operators.BlockingOperatorToFuture;
+import rx.internal.util.BlockingUtils;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * {@code BlockingSingle} is a blocking "version" of {@link Single} that provides blocking
+ * operators.
+ *
+ * You construct a {@code BlockingSingle} from a {@code Single} with {@link #from(Single)}
+ * or {@link Single#toBlocking()}.
+ */
+@Experimental
+public class BlockingSingle<T> {
+	private final Single<? extends T> single;
+
+	private BlockingSingle(Single<? extends T> single) {
+		this.single = single;
+	}
+	
+	/**
+	 * Converts a {@link Single} into a {@code BlockingSingle}.
+	 *
+	 * @param single the {@link Single} you want to convert
+
+	 * @return a {@code BlockingSingle} version of {@code single}
+	 */
+	public static <T> BlockingSingle<T> from(Single<? extends T> single) {
+		return new BlockingSingle<T>(single);
+	}
+
+	/**
+	 * Returns the item emitted by this {@code BlockingSingle}.
+	 *
+	 * If the underlying {@link Single} returns successfully, the value emitted
+	 * by the {@link Single} is returned. If the {@link Single} emits an error,
+	 * the throwable emitted ({@link SingleSubscriber#onError(Throwable)}) is
+	 * thrown.
+	 *
+	 *  @return the value emitted by this {@code BlockingSingle}
+	 */
+	public T value() {
+		final AtomicReference<T> returnItem = new AtomicReference<T>();
+		final AtomicReference<Throwable> returnException = new AtomicReference<Throwable>();
+		final CountDownLatch latch = new CountDownLatch(1);
+		Subscription subscription = single.subscribe(new SingleSubscriber<T>() {
+			@Override
+			public void onSuccess(T value) {
+				returnItem.set(value);
+				latch.countDown();
+			}
+
+			@Override
+			public void onError(Throwable error) {
+				returnException.set(error);
+				latch.countDown();
+			}
+		});
+
+		BlockingUtils.awaitForComplete(latch, subscription);
+		Throwable throwable = returnException.get();
+		if (throwable != null) {
+			if (throwable instanceof RuntimeException) {
+				throw (RuntimeException)throwable;
+			}
+			throw new RuntimeException(throwable);
+		}
+		return returnItem.get();
+	}
+
+	/**
+	 * Returns a {@link Future} representing the value emitted by this {@code BlockingSingle}.
+	 *
+	 * @return a {@link Future} that returns the value
+	 */
+	public Future<T> toFuture() {
+		return BlockingOperatorToFuture.toFuture(single.toObservable());
```

Comment:
Is this "appropriate"? Could add `BlockingOperatorToFuture.toFuture(Single)` - wonder if it would/should behave any differently.